lighthouse-core/gather/computed/page-dependency-graph.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ComputedArtifact = require('./computed-artifact');
+const Node = require('./dependency-graph/node');
+const Emulation = require('../../lib/emulation');
+
+class PageDependencyGraphArtifact extends ComputedArtifact {
+  get name() {
+    return 'PageDependencyGraph';
+  }
+
+  get requiredNumberOfArtifacts() {
+    return 2;
+  }
+
+  static getNetworkInitiators(record) {
+    if (!record._initiator) return [];
```

Comment:
I'll revisit using `initiatorRequest` when I flesh this out, but for now I was getting cyclic dependencies on certain runs